sygnal/utils.py

```diff
@@ -15,11 +15,15 @@
 import json
 import re
 from logging import LoggerAdapter
+from typing import TYPE_CHECKING, Any, MutableMapping, Pattern, Tuple, Union
 
 from twisted.internet.defer import Deferred
 
+if TYPE_CHECKING:
+    from sygnal.sygnal import SygnalReactor
 
-async def twisted_sleep(delay, twisted_reactor):
+
+async def twisted_sleep(delay: float, twisted_reactor: "SygnalReactor") -> None:
```

Comment:
This is fine as it is, but I wanted to mention a couple of alternatives here. One be to write `twisted_reactor: IReactorTime` and only import `IReactorTime`; a second would be to move `SygnalReactor` into a new file, say `sygnal/types.py`.

Reply:
Great, I appreciate knowing alternative solutions, it really helps to have more tools in my arsenal.